### PR TITLE
Fix/help modal fab button toggle issue 1681

### LINKED ISSUE #1681: [Bug] On opening the help modal from fab button, it must change to close button.
The fab button is not changing its state on opening. Ideally, it should change to a close button state.
fab-issue

Ideal UX:
fab-ideal

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #1681

### CODE REVIEW

**Comment by hetunandu** [comment]
@Nikhil-Nandagopal I hope the close button looks fine. Do we need to design this?

File: app/client/cypress/integration/Smoke_TestSuite/UnitTest/LoginFromUIApp_spec.js [hunk 1]
```diff
@@ -46,4 +47,12 @@ describe("Login from UI and check the functionality", function() {
     cy.get(homePage.profileMenu).click();
     cy.get(homePage.themeText).should("have.attr", "value", "true");
   });
+
+  it("Icon of fab button of help modal should change on open", function() {
+    cy.get(HelpLocators.HelpButton).click();
+    cy.get(`${HelpLocators.HelpButton} .bp3-icon-cross`).should(
+      "have.length",
+      1,
+    );
```
Comment (hetunandu) [inline app/client/cypress/integration/Smoke_TestSuite/UnitTest/LoginFromUIApp_spec.js:56]: Also test closing by clicking it

File: app/client/src/components/designSystems/appsmith/help/HelpModal.tsx [hunk 2]
```diff
@@ -76,44 +78,65 @@ class HelpModal extends React.Component<Props> {
...
             highlight={!isHelpModalOpen}
             layer={layers.help}
-            onClick={() => {
-              AnalyticsUtil.logEvent("OPEN_HELP", { page: this.props.page });
-              dispatch(setHelpModalVisibility(!isHelpModalOpen));
-            }}
+            onClick={this.onOpen}
           >
-            <HelpIcon />
+            {isHelpModalOpen ? <CloseIcon /> : <HelpIcon />}
```
Comment (hetunandu) [inline app/client/src/components/designSystems/appsmith/help/HelpModal.tsx:139]: Close icon does not have the right cursor on hover

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Comment by Nikhil-Nandagopal** [comment]
LGTM

File: app/client/cypress/integration/Smoke_TestSuite/UnitTest/LoginFromUIApp_spec.js (hunk 1 above)
Comment (jsartisan) [reply app/client/cypress/integration/Smoke_TestSuite/UnitTest/LoginFromUIApp_spec.js:56]: Done.

File: app/client/src/components/designSystems/appsmith/help/HelpModal.tsx (hunk 2 above)
Comment (jsartisan) [reply app/client/src/components/designSystems/appsmith/help/HelpModal.tsx]: Done.

File: app/client/cypress/integration/Smoke_TestSuite/UnitTest/LoginFromUIApp_spec.js (hunk 1 above)
Comment (jsartisan) [reply app/client/cypress/integration/Smoke_TestSuite/UnitTest/LoginFromUIApp_spec.js:56]: Done.

File: app/client/src/components/designSystems/appsmith/help/HelpModal.tsx (hunk 2 above)
Comment (jsartisan) [reply app/client/src/components/designSystems/appsmith/help/HelpModal.tsx]: Fixed the hover issue.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**